libmproxy/proxy/root_context.py

```diff
@@ -47,6 +74,7 @@ def next_layer(self, top_layer):
         layer = self._next_layer(top_layer)
         return self.channel.ask("next_layer", layer)
 
+    @tls_sni_check_ignore
```

Comment:
This decorator-approach feels a bit weird. 
Can we just do this explicitly in the `_next_layer` function? We can move the `is_tls_record_magic` part to the top, then check if `self.config.check_ignore or self.config.check_tcp` is True and then parse the ClientHello if so.